sparkviz/__init__.py

```diff
@@ -32,22 +35,27 @@ def special_show(self, n=2000, truncate=False, vertical=False, auto_sample=True,
     sampled_df = self
     if auto_sample:
       total_count = self.count()
-      if n < total_count:
+      do_sample = (n < total_count) and auto_sample
+      if do_sample:
         fraction = (n * 1.1) / total_count
         sampled_df = self.sample(withReplacement=False, fraction = fraction).limit(n)
       pandas_df = sampled_df.toPandas()
-    DataFrameResult(pandas_df, self)
+    DataFrameResult(pandas_df, self, do_sample)
   finally:
     if do_cache:
       df.unpersist()
 
 class DataFrameResult():
   """A special fake class for DataFrame results. Wraps a pandas df & a spakr df."""
 
-  def __init__(self, pdf, sdf):
+  def __init__(self, pdf, sdf, sampled):
     self.pdf = pdf
     self.sdf = sdf
+    self.sampled = sampled
     from IPython.display import display
+    # TODO(communicate if we've sampled data to the front end?)
     self.display = display(pdf, display_id=True)
```

Comment:
It just occurred to me that we could indicate sampled data in another way, which is `metadata` on the displayed result.

```
display(
    JSON({'a': 'b'}),
    metadata={
        "application/json": {
            "expanded": True
        }
    }
)
```

![screen shot 2017-12-22 at 8 13 17 am](https://user-images.githubusercontent.com/836375/34304778-fe8ae0f8-e6ef-11e7-8875-1e06c949d600.png)